nomad/nomad_deployer.go

```diff
@@ -169,3 +185,62 @@ func compareJobs(old, new *nomadapi.Job) bool {
 
 	return false
 }
+
+// addNomadDefaultEnv injects environment varibles inherited from Nomad environment
+func addNomadDefaultEnv(job *nomadapi.Job) {
+	task := job.TaskGroups[0].Tasks[0]
```

Comment:
Maybe it's a good idea to check the length of `TaskGroups` and `TaskGroups[0].Tasks` before this line?

Reply:
It's a good consideration. The check was added in this function but later extracted out which is in line 241-245